lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -682,52 +691,58 @@ def _post(self, params: ApiParams, _) -> Response:
         # `try` block to catch failures.
         try:
             # Pass the bulk command generator to the helper
-            results = helpers.streaming_bulk(
-                elastic,
-                self.generate_actions(params.body, dataset),
-                raise_on_exception=False,
-                raise_on_error=False,
-            )
+            actions = self.generate_actions(params.body, dataset)
+
+            # If generate_actions returns None then no Elasticsearch operations
+            # are required. (For example, the subclass finds no Elasticsearch
+            # documents.)
+            if actions:
```

Comment:
This will always be `True`:

```
def gen(do_nothing=False):
    if do_nothing:
        return None
    for x in [1, 2, 3]:
        yield x

print(repr(gen()))
print(repr(gen(do_nothing=True)))
```